### PR TITLE
DAOS-12079 test: Move HW Small tests to HW Medium

### PR DESCRIPTION
**Pull request by phender**
Replacing hw,small test tags with hw,medium.

Test-tag: DfuseSpaceCheck BoundaryPoolContainerSpace FullPoolContainerCreate ConfigGenerateOutput ConfigGenerateRun DaosPrivHelperTest DmgNetworkScanTest DmgPoolQueryTest DmgStorageQuery DmgStorageScanSCMTest DmgSystemReformatTest TestWithTelemetryIOBasic TestWithTelemetryNvme SSDSocketTest DaosPerf DaosVol DmvrCopyProcs DmvrNegativeSpaceTest Cmd DaosBuild RootContainerTest PosixSimul SparseFile HarnessAdvancedTest HarnessBasicTest HarnessSetupTest HarnessSlurmTest HarnessTimeoutTest IorInterceptBasic IorInterceptMessages Hdf5 Romio CPUUsage DaosServerConfigTest StorageTiers

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — DISMISSED** [review]

**Comment by phender** [comment]
In https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/1/ the following errors are expected as these tests verify that when errors are generated, they are correctly reported:

- [Test / Functional on EL 8 / no_cmocka_xml_file_test.4-./harness/basic.py:HarnessBasicTest.test_no_cmocka_xml;run-4712](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/1/testReport/junit/(root)/no_cmocka_xml_file_test/Test___Functional_on_EL_8___4___harness_basic_py_HarnessBasicTest_test_no_cmocka_xml_run_4712/)
- [Test / Functional on EL 8 / FTEST_harness.HarnessTimeoutTest.1-./harness/timeout.py:HarnessTimeoutTest.test_timeout;run-8998](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/1/testReport/junit/FTEST_harness/HarnessTimeoutTest/Test___Functional_on_EL_8___1___harness_timeout_py_HarnessTimeoutTest_test_timeout_run_8998/)
- [Test / Functional on EL 8 / FTEST_launch.harness-advanced.03-./harness/advanced.py](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/1/testReport/junit/FTEST_launch/harness-advanced/Test___Functional_on_EL_8___03___harness_advanced_py/)
- [Test Hardware / Functional Hardware Large / FTEST_harness.HarnessTimeoutTest.1-./harness/timeout.py:HarnessTimeoutTest.test_timeout_hw;run-8998](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/1/testReport/junit/FTEST_harness/HarnessTimeoutTest/Test_Hardware___Functional_Hardware_Large___1___harness_timeout_py_HarnessTimeoutTest_test_timeout_hw_run_8998/)
- [Test Hardware / Functional Hardware Large / FTEST_launch.harness-advanced.05-./harness/advanced.py](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/1/testReport/junit/FTEST_launch/harness-advanced/Test_Hardware___Functional_Hardware_Large___05___harness_advanced_py/)

This additional failure is also expected as slurm_setup.py was not run due to the lack of a `soak` tag:
- [Test Hardware / Functional Hardware Large / FTEST_launch.harness-slurm.02-./harness/slurm.py](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/1/testReport/junit/FTEST_launch/harness-slurm/Test_Hardware___Functional_Hardware_Large___02___harness_slurm_py/)

**Review by daltonbohning — COMMENTED** [review]
Why move some tests to large instead of medium?

**Review by tanabarr — DISMISSED** [review]
LGTM, including specifically the changes to ssd_socket.py

**Comment by phender** [comment]
In https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/3/ all tests passed except the `container/full_pool_container_create.py` tests which should be skipped.  They currently are run because the `/scratch/CI-skip-list-master` skip list contains three entries for this test, where the first match is resolved:
```
[['DAOS-7374', 'test_method_name', 'test_no_space_cont_create']]|137fe28
[['DAOS-8400', 'test_method_name', 'test_no_space_cont_create']]
[['DAOS-5813', 'test_method_name', 'test_no_space_cont_create']]
```
A simple solution would be to remove this resolved entry and the test would be skipped again.

**Comment by phender** [comment]
https://daosio.atlassian.net/browse/DAOS-12139 has been created for the skip-list entry and the skip-lists have been updated to re-skip the `container/full_pool_container_create.py` tests

**Comment by phender** [comment]
> Why move some tests to large instead of medium?

The HW Medium clusters use VMD, which causes those tests to fail, so instead they were moved to the HW Large clusters which do not use VMD.

**Review by daltonbohning — DISMISSED** [review]

**Review by daltonbohning — DISMISSED** [review]

**Review by tanabarr — APPROVED** [review]

**Comment by phender** [comment]
Failures in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/5

- daos_vol/h5_suite.py: new issue https://daosio.atlassian.net/browse/DAOS-12144
- ior/intercept_basic.py: known issue https://daosio.atlassian.net/browse/DAOS-11830

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
The failed tests from https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/5 where re-run in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10851/6/ and they passed

**Review by mjean308 — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

File: src/tests/ftest/control/dmg_telemetry_nvme.py [hunk 1]
```diff
@@ -42,13 +42,12 @@ def test_nvme_telemetry_metrics(self):
...
-            "ENGINE_NVME_TEMP_METRICS",
-            "ENGINE_NVME_TEMP_TIME_METRICS",
-            "ENGINE_NVME_RELIABILITY_METRICS",
-            "ENGINE_NVME_INTEL_VENDOR_METRICS"]
+        groups = ["ENGINE_NVME_HEALTH_METRICS",
+                  "ENGINE_NVME_CRIT_WARN_METRICS",
+                  "ENGINE_NVME_TEMP_METRICS",
+                  "ENGINE_NVME_TEMP_TIME_METRICS",
+                  "ENGINE_NVME_RELIABILITY_METRICS",
+                  "ENGINE_NVME_INTEL_VENDOR_METRICS"]
```
Comment (daltonbohning) [inline src/tests/ftest/control/dmg_telemetry_nvme.py:50]: FWIW, I think the old format is generally better

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**